src/middlewares/auth.js

```diff
@@ -0,0 +1,29 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+
+dotenv.config();
+const secret = process.env.JWTSECRET;
+
+const verifyUser = (req, res, next) => {
+  const { token } = req.headers;
+  try {
+    if (!token) {
+      return res.status(403).json({
+        status: 403,
+        error: 'Authentication failed, please login',
+      });
+    }
+    const decoded = jwt.verify(token, secret);
+    if (decoded) {
```

Comment:
What happens if the token has expired?

Reply:
The catch block handles that by throwing error access denied